JunList.java

```diff
@@ -19,99 +21,104 @@ public void append(int value) {
+        trackListPosition += 1;
+    }
 
-        public void add(int position,int value){
-            //creates list twice as big when necessary
-            if (trackListPosition == list.length) {
-                int[] newList = new int[list.length * 2];
-                for (int i = 0; i < list.length; ++i) {
-                    if (i < position) {
-                        newList[i] = list[i];
-                    } else if (position == i) {
-                        newList[i] = value;
-                    } else {
-                        newList[i] = list[i - 1];
-                    }
-                }
-                this.list = newList;
-            } else {
-                for (int i = trackListPosition; i > position; i--) {
-                    list[i] = list[i - 1];
+    public int get(int index) {
+        return this.list[index];
+    }
+
+    public void add(int position, int value) {
+        //creates list twice as big when necessary
+        if (trackListPosition == list.length) {
+            int[] newList = new int[list.length * 2];
+            for (int i = 0; i < list.length; ++i) {
+                if (i < position) {
+                    newList[i] = list[i];
+                } else if (position == i) {
+                    newList[i] = value;
+                } else {
+                    newList[i] = list[i - 1];
                 }
-                this.list[position] = value;
             }
-            trackListPosition += 1;
+            this.list = newList;
+        } else {
+            for (int i = trackListPosition; i > position; i--) {
+                list[i] = list[i - 1];
+            }
+            this.list[position] = value;
         }
+        trackListPosition += 1;
+    }
 
-        public void prepend(int value){
-            //creates list twice as big when necessary
-            if (trackListPosition == list.length) {
-                int[] newList = new int[list.length * 2];
-                for (int i = 0; i < list.length; i++) {
-                    if (i == 0) {
-                        newList[i] = value;
-                    } else {
-                        newList[i] = list[i - 1];
-                    }
+    public void prepend(int value) {
+        //creates list twice as big when necessary
+        if (trackListPosition == list.length) {
+            int[] newList = new int[list.length * 2];
+            for (int i = 0; i < list.length + 1; i++) {
+                if (i == 0) {
+                    newList[i] = value;
+                } else {
+                    newList[i] = list[i - 1];
                 }
-                this.list = newList;
-            } else {
-                for (int i = trackListPosition; i >= 0; i--) {
-                    if (i == 0) {
-                        list[i] = value;
-                    } else {
-                        list[i] = list[i - 1];
-                    }
+            }
+            this.list = newList;

```

Comment:
you don't need the this keyword, really at all in this these files. Not something you could've known, but can you not add new at the start of variable of names like newList, the reason is that creating objects looks like Object name = new Object(); If you wrote Object blah = new newxx() it'd be weird-ish.

Reply:
ok i removed this keyword and changed newList / newQueue / newStack to temporaryList/temporaryQueue/temporaryStack